pkg/service/generator.go

```diff
@@ -232,61 +231,42 @@ func NewEmqxStatefulSet(emqx v1beta1.Emqx, labels map[string]string, ownerRefs [
 	return sts
 }
 
-func generateInitContainers(emqx v1beta1.Emqx) []corev1.Container {
-	runAsUser := int64(0)
-	privileged := true
-	return []corev1.Container{
-		{
-			Image: emqx.GetImage(),
-			Name:  "emqx-volume-change-ownership",
-			Command: []string{
-				"/bin/bash",
-				"-c",
-				"mkdir -p /opt/emqx/data/ && mkdir -p /opt/emqx/log &&chown -R 1000:1000 /opt/emqx/data && chown -R 1000:1000 /opt/emqx/log",
-			},
-			SecurityContext: &corev1.SecurityContext{
-				RunAsUser:  &runAsUser,
-				Privileged: &privileged,
-			},
-			VolumeMounts: []corev1.VolumeMount{
-				{
-					Name:      emqx.GetDataVolumeName(),
-					MountPath: constants.EMQX_DATA_DIR,
-				},
-				{
-					Name:      emqx.GetLogVolumeName(),
-					MountPath: constants.EMQX_LOG_DIR,
-				},
-			},
-		},
+func generateSecurityContext() *corev1.PodSecurityContext {
+	emqxUserGroup := int64(1000)
+	runAsNonRoot := true
+	fsGroupChangeAlways := corev1.FSGroupChangeAlways
+
+	return &corev1.PodSecurityContext{
+		FSGroup:             &emqxUserGroup,
+		FSGroupChangePolicy: &fsGroupChangeAlways,
+		RunAsNonRoot:        &runAsNonRoot,
+		RunAsUser:           &emqxUserGroup,
+		SupplementalGroups:  []int64{emqxUserGroup},
 	}
 }
-
-func getContainerSecurityContext() *corev1.SecurityContext {
+func generateContainerSecurityContext() *corev1.SecurityContext {
 	emqxUserGroup := int64(1000)
-	emqxUser := int64(1000)
```

Comment:
Maybe retain the emqxUser instead of emqxUserGroup to be set as the value of RunAsUser?

Reply:
Done